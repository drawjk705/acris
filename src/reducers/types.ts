import { Borough } from './constants';
export type TDocumentId = string;

export type TProperty = {
    bble: string;
    borough: Borough;
    block: number;
    lot: number;
    streetNumber: string;
    streetName: string;
    unit: string | undefined;
    propertyType: string;
    documentId: TDocumentId;
};

export type TPropertyType = {
    recordType: string;
    propertyType: string;
    description: string;
};

export type TDocument = {
    id: TDocumentId;
    crfn: string;
    type: string;
    date: string;
    amount: string;
};

export type TDocumentType = {
    typeId: string;
    documentType: string;
    classCodeDescription: string;
    partyOneType: string;
    partyTwoType: string;
    partyThreeType: string;
};

export type TParty = {
    partyType: String;
    documentId: TDocumentId;
    name: string | undefined;
    address: TAddress;
};

export type THousingMaintenanceCodeViolation = {
    apartment: string;
    story: string;
    inspectionDate: string;
    originalCertifyByDate: string;
    originalCorrectByDate: string;
    newCertifyByDate: string;
    newCorrectByDate: string;
    orderNumber: string;
    novDescription: string;
    novIssuedDate: string;
    currentStatus: string;
    communityBoard: string;
    violationStatus: string;
};

export type THpdJurisdictionData = {
    registrationId: number;
    buildingId: number;
    communityBoardId: number;
    managementProgram: string;
    numberOfLegalStories: number;
    numberOfApartments: number;
    numberOfRooms: number;
    lifecycleStage: string;
    recordStatusId: number;
    recordStatus: HpdJurisdictionRecordStatus;
};

export type TRegistrationContact = {
    type: string;
    contactDescription: string;
    corporationName: string;
    title: string;
    firstName: string;
    middleInitial: string;
    lastName: string;
    businessAddress: TAddress;
};

export type TValuationAndAssessmentData = {
    numberOfStories: number;
    marketValue: number;
    actualLandValue: number;
    actualTotalValue: number;
    actualExemptLandValue: number;
    actualExemptLandTotal: number;
    exceptionCodeOne: number;
    transitionalLandValue: number;
    transitionalTotalValue: number;
    transitionalExemptionLandValue: number;
    transitionalExemptionLandTotal: number;
    assessmentYear: string;
};

export type TTaxClassData = {
    zoningCode: string;
    dimensions: {
        lotFrontageFeet: number;
        lotDepthFeet: number;
        buildingFrontageFeet: number;
        buildingDepth: number;
        totalLandArea: number;
        grossSquarefootage: number;
        hotelAreaGrossSquarefootage: number;
        officeAreaGrossSquarefootage: number;
        residentialAreaGrossSquarefootage: number;
        retailAreaGrossSquarefootage: number;
        loftAreaGrossSquarefootage: number;
        factoryAreaGrossSquarefootage: number;
        warehouseAreaGrossSquarefootage: number;
        storageAreaGrossSquarefootage: number;
        garageAreaGrossSquarefootage: number;
        otherAreaGrossSquarefootage: number;
    };
    hasExtension: boolean;
    storyCount: number;
    unitCount: number;
    coopApartmentCount: number;
    coopIdNumber: number;
    buildingConstructionYear: number;
    assessmentAndExemptionValuations: {
        market?: AssessmentAndExemptionValuations;
        actual?: AssessmentAndExemptionValuations;
        transitional?: AssessmentAndExemptionValuations;
        taxable?: AssessmentAndExemptionValuations;
        tentativeMarket?: AssessmentAndExemptionValuations;
        tentativeActual?: AssessmentAndExemptionValuations;
        tentativeTransitional?: AssessmentAndExemptionValuations;
        tentativeTaxable?: AssessmentAndExemptionValuations;
        changeByNoticeMarket?: AssessmentAndExemptionValuations;
        changeByNoticeActual?: AssessmentAndExemptionValuations;
        changeByNoticeTransitional?: AssessmentAndExemptionValuations;
        changeByNoticeTaxable?: AssessmentAndExemptionValuations;
        finalMarket?: AssessmentAndExemptionValuations;
        finalActual?: AssessmentAndExemptionValuations;
        finalTransitional?: AssessmentAndExemptionValuations;
        finalTaxable?: AssessmentAndExemptionValuations;
        currentMarket?: AssessmentAndExemptionValuations;
        currentActual?: AssessmentAndExemptionValuations;
        currentTransitional?: AssessmentAndExemptionValuations;
        currentTaxable?: AssessmentAndExemptionValuations;
    };
};

export type AssessmentAndExemptionValuations = {
    assessment?: Valuation;
    exemption?: Valuation;
};

export type Valuation = {
    landValue?: number;
    totalValue?: number;
};

export type TAddress = {
    addressLineOne: string;
    addressLineTwo: string;
    city: string;
    state: string;
    zipCode: string;
};

export enum ViolationCurrentStatus {
    CERTIFICATION_POSTPONMENT_DENIED = 'CERTIFICATION POSTPONMENT DENIED',
    CERTIFICATION_POSTPONMENT_GRANTED = 'CERTIFICATION POSTPONMENT GRANTED',
    CIV14_MAILED = 'CIV14 MAILED',
    COMPLIED_IN_ACCESS_AREA = 'COMPLIED IN ACCESS AREA',
    DEFECT_LETTER_ISSUED = 'DEFECT LETTER ISSUED',
    FALSE_CERTIFICATION = 'FALSE CERTIFICATION',
    FIRST_NO_ACCESS_TO_RE_INSPECT_VIOLATION = 'FIRST NO ACCESS TO RE- INSPECT VIOLATION',
    INFO_NOV_SENT_OUT = 'INFO NOV SENT OUT',
    INVALID_CERTIFICATION = 'INVALID CERTIFICATION',
    NOT_COMPLIED_WITH = 'NOT COMPLIED WITH',
    NOTICE_OF_ISSUANCE_SENT_TO_TENANT = 'NOTICE OF ISSUANCE SENT TO TENANT',
    NOV_CERTIFIED_LATE = 'NOV CERTIFIED LATE',
    NOV_CERTIFIED_ON_TIME = 'NOV CERTIFIED ON TIME',
    NOV_SENT_OUT = 'NOV SENT OUT',
    SECOND_NO_ACCESS_TO_RE_INSPECT_VIOLATION = 'SECOND NO ACCESS TO RE-INSPECT VIOLATION',
    VIOLATION_CLOSED = 'VIOLATION CLOSED',
    VIOLATION_DISMISSED = 'VIOLATION DISMISSED',
    VIOLATION_OPEN = 'VIOLATION OPEN',
    VIOLATION_REOPEN = 'VIOLATION REOPEN',
    VIOLATION_WILL_BE_REINSPECTED = 'VIOLATION WILL BE REINSPECTED',
}

export enum ViolationStatus {
    Open = 'Open',
    Close = 'Close',
}

export enum HpdJurisdictionRecordStatus {
    Active,
    Inactive,
    Pending,
}

export type AcrisType =
    | TProperty
    | TPropertyType
    | TDocument
    | TDocumentType
    | TParty
    | THousingMaintenanceCodeViolation
    | THpdJurisdictionData
    | TRegistrationContact
    | TValuationAndAssessmentData
    | TTaxClassData;
