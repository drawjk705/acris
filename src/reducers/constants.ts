export enum Borough {
    MANHATTAN = 'manhattan',
    BRONX = 'bronx',
    BROOKLYN = 'brooklyn',
    QUEENS = 'queens',
    STATEN_ISLAND = 'statenIsland',
}

export const BOROUGHS_BY_NAME: { [key in Borough]: number } = {
    [Borough.MANHATTAN]: 1,
    [Borough.BRONX]: 2,
    [Borough.BROOKLYN]: 3,
    [Borough.QUEENS]: 4,
    [Borough.STATEN_ISLAND]: 5,
};

export const BOROUGHS_BY_ID: { [key: number]: Borough } = {
    1: Borough.MANHATTAN,
    2: Borough.BRONX,
    3: Borough.BROOKLYN,
    4: Borough.QUEENS,
    5: Borough.STATEN_ISLAND,
};
