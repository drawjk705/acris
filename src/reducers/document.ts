import { TDocument } from './types';

export const reduceDocument = (documentObj: any): TDocument => ({
    id: documentObj.document_id,
    CRFN: documentObj.crfn,
    type: documentObj.doc_type,
    date: documentObj.document_date,
    amount: documentObj.document_amt,
});
