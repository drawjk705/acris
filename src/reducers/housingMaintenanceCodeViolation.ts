import { THousingMaintenanceCodeViolation } from './types';

export const reduceHousingMaintenanceCodeViolation = (
    violationObj: any
): THousingMaintenanceCodeViolation => ({
    apartment: violationObj.apartment,
    story: violationObj.story,
    inspectionDate: violationObj.inspectiondate,
    originalCertifyByDate: violationObj.originalcertifybydate,
    originalCorrectByDate: violationObj.originalcorrectbydate,
    orderNumber: violationObj.ordernumber,
    novDescription: violationObj.novdescription,
    novIssuedDate: violationObj.novissueddate,
    currentStatus: violationObj.currentstatus,
    communityBoard: violationObj.communityboard,
    newCertifyByDate: violationObj.newcertifybydate,
    newCorrectByDate: violationObj.newcorrectbydate,
    violationStatus: violationObj.violationstatus,
});
