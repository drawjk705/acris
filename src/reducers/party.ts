import { TParty } from './types';

export const reduceParty = (partyObj: any): TParty => ({
    documentId: partyObj.document_id,
    name: partyObj.name,
    addressLineOne: partyObj.address_1,
    addressLineTwo: partyObj.address_2,
    city: partyObj.city,
    state: partyObj.state,
    zipCode: partyObj.zip,
});
