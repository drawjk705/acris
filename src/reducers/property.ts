import { BOROUGHS_BY_ID } from './constants';
import { TProperty, TPropertyType } from './types';

export const reduceProperty = (propertyObj: any = {}): TProperty => ({
    borough: BOROUGHS_BY_ID[propertyObj.borough as number],
    block: propertyObj.block,
    lot: propertyObj.lot,
    streetNumber: propertyObj.street_number,
    streetName: propertyObj.street_name,
    unit: propertyObj.unit,
    propertyType: propertyObj.property_type,
    documentId: propertyObj.document_id,
});

export const reducePropertyType = (
    propertyTypeObj: any = {}
): TPropertyType => ({
    recordType: propertyTypeObj.record_type,
    propertyType: propertyTypeObj.property_type,
    description: propertyTypeObj.description,
});
