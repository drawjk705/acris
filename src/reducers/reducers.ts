import { BOROUGHS_BY_ID } from './constants';
import {
    TProperty,
    TPropertyType,
    TParty,
    THousingMaintenanceCodeViolation,
    TDocument,
} from './types';

export const Reducers = {
    reduceProperty: (propertyObj: any = {}): TProperty => ({
        borough: BOROUGHS_BY_ID[propertyObj.borough as number],
        block: propertyObj.block,
        lot: propertyObj.lot,
        streetNumber: propertyObj.street_number,
        streetName: propertyObj.street_name,
        unit: propertyObj.unit,
        propertyType: propertyObj.property_type,
        documentId: propertyObj.document_id,
    }),

    reducePropertyType: (propertyTypeObj: any = {}): TPropertyType => ({
        recordType: propertyTypeObj.record_type,
        propertyType: propertyTypeObj.property_type,
        description: propertyTypeObj.description,
    }),

    reduceParty: (partyObj: any = {}): TParty => ({
        documentId: partyObj.document_id,
        name: partyObj.name,
        addressLineOne: partyObj.address_1,
        addressLineTwo: partyObj.address_2,
        city: partyObj.city,
        state: partyObj.state,
        zipCode: partyObj.zip,
    }),

    reduceHousingMaintenanceCodeViolation: (
        violationObj: any = {}
    ): THousingMaintenanceCodeViolation => ({
        apartment: violationObj.apartment,
        story: violationObj.story,
        inspectionDate: violationObj.inspectiondate,
        originalCertifyByDate: violationObj.originalcertifybydate,
        originalCorrectByDate: violationObj.originalcorrectbydate,
        orderNumber: violationObj.ordernumber,
        novDescription: violationObj.novdescription,
        novIssuedDate: violationObj.novissueddate,
        currentStatus: violationObj.currentstatus,
        communityBoard: violationObj.communityboard,
        newCertifyByDate: violationObj.newcertifybydate,
        newCorrectByDate: violationObj.newcorrectbydate,
        violationStatus: violationObj.violationstatus,
    }),

    reduceDocument: (documentObj: any = {}): TDocument => ({
        id: documentObj.document_id,
        CRFN: documentObj.crfn,
        type: documentObj.doc_type,
        date: documentObj.document_date,
        amount: documentObj.document_amt,
    }),
};
