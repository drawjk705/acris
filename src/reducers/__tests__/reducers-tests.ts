import { Reducers } from '../reducers';
import { mocks } from './mockObjects';

describe('Reducers', () => {
    describe('reduceProperty', () => {
        it('reduces correctly', () => {
            const property = Reducers.reduceProperty(mocks.property);

            expect(property).toMatchSnapshot();
        });
    });

    describe('reducePropertyType', () => {
        it('reduces correctly', () => {
            const propertyType = Reducers.reducePropertyType(
                mocks.propertyType
            );

            expect(propertyType).toMatchSnapshot();
        });
    });

    describe('reduceHousingMaintenanceCodeViolation', () => {
        it('reduces correctly', () => {
            const violation = Reducers.reduceHousingMaintenanceCodeViolation(
                mocks.violation
            );

            expect(violation).toMatchSnapshot();
        });
    });

    describe('reduceDocument', () => {
        it('reduces correctly', () => {
            const document = Reducers.reduceDocument(mocks.document);

            expect(document).toMatchSnapshot();
        });
    });

    describe('reduceHpdJurisdictionData', () => {
        it('reduces correctly', () => {
            const result = Reducers.reduceHpdJurisdictionData(
                mocks.jurisdictionData
            );

            expect(result).toMatchSnapshot();
        });
    });

    describe('reduceRegistrationContact', () => {
        it('reduces correctly', () => {
            const result = Reducers.reduceRegistrationContact(mocks.contact);

            expect(result).toMatchSnapshot();
        });
    });

    describe('reduceValuationAndAssessmentData', () => {
        it('reduces correctly', () => {
            const result = Reducers.reduceValuationAndAssessmentData(
                mocks.valuationAndAssessmentData
            );

            expect(result).toMatchSnapshot();
        });
    });

    describe('reduceTaxClassData', () => {
        it('reduces correctly', () => {
            const result = Reducers.reduceTaxClassData(mocks.taxClassData);

            expect(result).toMatchSnapshot();
        });
    });

    describe('reduceDocumentType', () => {
        it('reduces correctly', () => {
            const result = Reducers.reduceDocumentType(mocks.documentType);

            expect(result).toMatchSnapshot();
        });
    });
});
