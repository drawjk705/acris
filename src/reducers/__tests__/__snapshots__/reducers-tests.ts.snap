// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers reduceDocument reduces correctly 1`] = `
Object {
  "amount": "485000.00",
  "crfn": "2020000144847",
  "date": "2020-04-10T00:00:00.000",
  "id": "2020050800759005",
  "type": "AGMT",
}
`;

exports[`Reducers reduceDocumentType reduces correctly 1`] = `
Object {
  "classCodeDescription": "UCC AND FEDERAL LIENS",
  "documentType": "INITIAL COOP UCC1",
  "partyOneType": "DEBTOR",
  "partyThreeType": "ASSIGNEE",
  "partyTwoType": "SECURED PARTY",
  "typeId": "INIC",
}
`;

exports[`Reducers reduceHousingMaintenanceCodeViolation reduces correctly 1`] = `
Object {
  "apartment": undefined,
  "communityBoard": "1",
  "currentStatus": "VIOLATION DISMISSED",
  "inspectionDate": "1997-04-11T00:00:00.000",
  "newCertifyByDate": undefined,
  "newCorrectByDate": undefined,
  "novDescription": "SECTION 27-2098 ADM CODE  FILE WITH THIS DEPARTMENT A REGISTRATION STATEMENT FOR BUILDING.",
  "novIssuedDate": "1997-04-22T00:00:00.000",
  "orderNumber": "772",
  "originalCertifyByDate": "1997-08-15T00:00:00.000",
  "originalCorrectByDate": "1997-08-08T00:00:00.000",
  "story": "All Stories",
  "violationStatus": "Close",
}
`;

exports[`Reducers reduceHpdJurisdictionData reduces correctly 1`] = `
Object {
  "buildingId": 859129,
  "communityBoardId": 0,
  "lifecycleStage": "Building",
  "managementProgram": "PVT",
  "numberOfApartments": NaN,
  "numberOfLegalStories": NaN,
  "numberOfRooms": NaN,
  "recordStatus": "Inactive",
  "recordStatusId": 2,
  "registrationId": 0,
}
`;

exports[`Reducers reduceParty reduces correctly 1`] = `
Object {
  "address": Object {
    "addressLineOne": "175 PINELAWN ROAD",
    "addressLineTwo": "SUITE 400",
    "city": "MELVILLE",
    "state": "NY",
    "zipCode": "11747",
  },
  "documentId": "2018092100197002",
  "name": "FREEDOM MORTGAGE CORPORATION",
  "partyType": "2",
}
`;

exports[`Reducers reduceProperties reduces correctly 1`] = `
Object {
  "bble": "1005110001",
  "block": 511,
  "borough": 1,
  "documentIds": Array [
    "2016010401364001",
    "2012091700987004",
  ],
  "lot": 1,
  "propertyType": "OF",
  "streetName": "BROADWAY",
  "streetNumber": "568",
  "unit": undefined,
}
`;

exports[`Reducers reducePropertyType reduces correctly 1`] = `
Object {
  "description": "ADJACENT CONDOMINIUM UNIT TO BE COMBINED",
  "propertyType": "CA",
  "recordType": "G",
}
`;

exports[`Reducers reduceRegistrationContact reduces correctly 1`] = `
Object {
  "businessAddress": Object {
    "addressLineOne": "312 W 109TH ST",
    "addressLineTwo": "Apartment 102",
    "city": "NEW YORK",
    "state": "NY",
    "zipCode": "10025",
  },
  "contactDescription": "CORP",
  "corporationName": "ACADEMY STUDOS",
  "firstName": "SVEIN",
  "lastName": "JORGENSEN",
  "middleInitial": undefined,
  "title": undefined,
  "type": "Agent",
}
`;

exports[`Reducers reduceTaxClassData reduces correctly 1`] = `
Object {
  "assessmentAndExemptionValuations": Object {
    "actual": Object {
      "assessment": Object {
        "landValue": 387450,
        "totalValue": 3280500,
      },
      "exemption": Object {
        "landValue": 0,
      },
    },
    "changeByNoticeActual": Object {
      "assessment": Object {
        "landValue": 387450,
        "totalValue": 3398400,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "changeByNoticeMarket": Object {
      "assessment": Object {
        "landValue": 861000,
        "totalValue": 7552000,
      },
    },
    "changeByNoticeTaxable": Object {
      "assessment": Object {
        "totalValue": 2976300,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "changeByNoticeTransitional": Object {
      "assessment": Object {
        "landValue": NaN,
        "totalValue": 2976300,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "currentActual": Object {
      "assessment": Object {
        "landValue": 387450,
        "totalValue": 3398400,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "currentMarket": Object {
      "assessment": Object {
        "landValue": 861000,
        "totalValue": 7552000,
      },
    },
    "currentTaxable": Object {
      "assessment": Object {
        "totalValue": 2976300,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "currentTransitional": Object {
      "assessment": Object {
        "landValue": 387450,
        "totalValue": 2976300,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "finalActual": Object {
      "assessment": Object {
        "landValue": 387450,
        "totalValue": 3398400,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "finalMarket": Object {
      "assessment": Object {
        "landValue": 861000,
        "totalValue": 7552000,
      },
    },
    "finalTaxable": Object {
      "assessment": Object {
        "totalValue": "2976300",
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "finalTransitional": Object {
      "assessment": Object {
        "landValue": 387450,
        "totalValue": 2976300,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "market": Object {
      "assessment": Object {
        "landValue": 861000,
        "totalValue": 7290000,
      },
    },
    "taxable": Object {
      "assessment": Object {
        "totalValue": 2751620,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "tentativeActual": Object {
      "assessment": Object {
        "landValue": 387450,
        "totalValue": 3398400,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "tentativeMarket": Object {
      "assessment": Object {
        "landValue": 861000,
        "totalValue": 7552000,
      },
    },
    "tentativeTaxable": Object {
      "assessment": Object {
        "totalValue": 2976300,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "tentativeTransitional": Object {
      "assessment": Object {
        "landValue": 387450,
        "totalValue": 2976300,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
    "transitional": Object {
      "assessment": Object {
        "landValue": 387450,
        "totalValue": 2751620,
      },
      "exemption": Object {
        "totalValue": 0,
      },
    },
  },
  "buildingConstructionYear": "1900",
  "coopApartmentCount": NaN,
  "coopIdNumber": 0,
  "dimensions": Object {
    "buildingDepth": 90,
    "buildingFrontageFeet": 44,
    "factoryAreaGrossSquarefootage": 0,
    "garageAreaGrossSquarefootage": 0,
    "grossSquarefootage": 23760,
    "hotelAreaGrossSquarefootage": 0,
    "loftAreaGrossSquarefootage": 0,
    "lotDepthFeet": 100,
    "lotFrontageFeet": 44,
    "officeAreaGrossSquarefootage": 0,
    "otherAreaGrossSquarefootage": 0,
    "residentialAreaGrossSquarefootage": 20760,
    "retailAreaGrossSquarefootage": 3000,
    "storageAreaGrossSquarefootage": 0,
    "totalLandArea": 4433,
    "warehouseAreaGrossSquarefootage": 0,
  },
  "hasExtension": false,
  "storyCount": 6,
  "unitCount": 26,
  "zoningCode": "C2-7A",
}
`;

exports[`Reducers reduceValuationAndAssessmentData reduces correctly 1`] = `
Object {
  "actualExemptLandTotal": 0,
  "actualExemptLandValue": 0,
  "actualLandValue": 5563,
  "actualTotalValue": 6924,
  "assessmentYear": "2018/19",
  "exceptionCodeOne": NaN,
  "marketValue": 229000,
  "numberOfStories": 0.5,
  "transitionalExemptionLandTotal": NaN,
  "transitionalExemptionLandValue": NaN,
  "transitionalLandValue": NaN,
  "transitionalTotalValue": NaN,
}
`;
